sdk/storage/src/common/shared_key_policy.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "common/shared_key_policy.hpp"
+
+#include "common/crypt.hpp"
+#include "common/storage_url_builder.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <iostream>
+
+namespace Azure { namespace Storage {
+  std::string SharedKeyPolicy::GetSignature(const Core::Http::Request& request) const
+  {
+    std::string string_to_sign;
+    string_to_sign += Azure::Core::Http::HttpMethodToString(request.GetMethod()) + "\n";
+
+    const auto& headers = request.GetHeaders();
+    for (std::string headerName :
+         {"Content-Encoding",
+          "Content-Language",
+          "Content-Length",
+          "Content-MD5",
+          "Content-Type",
+          "Date",
+          "If-Modified-Since",
+          "If-Match",
+          "If-None-Match",
+          "If-Unmodified-Since",
+          "Range"})
```

Comment:
Just curious. Do foreach in C++ guarantee sequence for array?

Reply:
i think it's an initialization list